sdk/core/azure_core_amqp/src/error.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corp. All Rights Reserved.
+// cspell: words amqp
+
+#[cfg(all(feature = "enable-fe2o3-amqp", not(target_arch = "wasm32")))]
+use crate::fe2o3::error::Fe2o3AmqpError;
+
+pub enum ErrorKind {
+    #[cfg(all(feature = "enable-fe2o3-amqp", not(target_arch = "wasm32")))]
+    IronOxideError { source: Fe2o3AmqpError },
+    #[cfg(any(not(feature = "enable-fe2o3-amqp"), target_arch = "wasm32"))]
+    NoopError,
+}
+
+pub struct AmqpError {
```

Comment:
It's not common - though not necessarily a bad thing, @JeffreyRichter? - to use anything other than `Error` since it's already namespaced to your crate. Users might use `azure_core_amqp::Error` or `use azure_core_amqp::Error as AmqpError` if they really need to differentiate.

Also, can we not just use what is in `typespec::error::Error` for this? Your factory functions below can be just that: functions. I did the same for HTTP errors so I could keep HTTP support conditional.

Reply:
I'll move it to Error.